packages/docker-engine/docker-engine.spec

```diff
@@ -29,11 +29,8 @@ BuildRequires: git
 BuildRequires: %{_cross_os}glibc-devel
 BuildRequires: %{_cross_os}libseccomp-devel
 BuildRequires: %{_cross_os}systemd-devel
-Requires: %{_cross_os}containerd
```

Comment:
I chose this line to ask a general question. Are all of these `Requires:` removals actually removing something that isn't required? In this example, isn't containerd actually required for Docker to work?

If that's the case (i.e. if some of the `Requires:` are _actually_ required), would it be better to leave them in the spec, comment them out in the Cargo.toml, and note in the Cargo.toml that they are commented out for dependency graph optimization?

If you are trying to compose a new variant, you would seemingly get an error message from the build rather than a missing dependency at runtime if your new variant's dependency graph happened to not build all the required packages.

Reply:
I've reworked the commits to do as you suggest - keep the `Requires:` while commenting out the dependency.